app/actions/bonds.ts

```diff
@@ -386,6 +336,16 @@ export const calculateUserBondDetails = (params: {
     }
 
     // Calculate bond details.
+    const bondContract = contractForBond({
+      bond: params.bond,
+      provider: params.provider,
+    });
+
+    const reserveContract = contractForReserve({
+      bond: params.bond as BondWithoutInverse, // we know here that it can't be inverse_usdc
```

Comment:
since we do this often it would be cool if ts was smart enough to figure that out on it's own. You can do something like this:

```
const getIsInverse = (bond: NormalBond | InverseBond): bond is InverseBond => {
  return bond === "inverse_usdc";
}
```

Reply:
Good idea, done with Exclude and Extract from TS ⭐